### ISSUE TITLE
DST issue in npgsql bulk upload with TSTZRange data type for Postrgre

### ISSUE THREAD

**Issue by Shane32** [linq2db/linq2db#2395]
When bulk inserting data into postgre, with the `tstzrange` date type, npgsql native bulk upload ignores the exact time of day when converting from UTC.

This causes conversion errors for any date that occurs on a DST changeover day after the changeover time.

To test this issue on a computer, bulk insert data of the `tstzrange` data type where one of the dates consist of a DST changeover date with a time of midnight.  It will convert successfully.  Then try with 4am; it will fail to convert successfully.

Note: don't test fall dates using times between 1am and 2am, since they occur twice due to DST (so those times represent two possible UTC times).  The .Net `TimeZoneInfo` class assumes 1am is after the change, not before it.

**Comment by Shane32** [comment]
Note that during the bulk upload of the tstzrange values, dates of a DateTimeKind property set to Unspecified are interpreted as Utc, whereas data returned from a select query is in the local time zone.

**Comment by Shane32** [comment]
Ideally `tstzrange` should be of the type `NpgsqlRange<DateTimeOffset>`, which is not yet supported by npgsql.  See here:
* https://github.com/npgsql/npgsql/issues/2436 - Support range with more than one CLR type
* https://github.com/npgsql/npgsql/pull/2438 - Range cleanup and support range with two CLR types

This still doesn't fix the conversion bug within npgsql for DateTime types, but at least people would have a bug-free option to set and return values.

**Comment by Shane32** [comment]
Thanks to @MaceWindu I've realized that I misinterpreted the failing test.  Bulk uploading data interprets dates with DateTimeKind set to Unspecified as Utc; so it is inserting Utc dates and returning Local times.  Hence, the changeover times I tested were off by 5 hours (eastern time).  Once the test is reconfigured to insert Utc times at the changeover time, and compares the results with those times converted to local time, all tests pass successfully.

Seems like unintuitive behavior, but that's what it does.

See here, where it by default returns times in local time:
https://github.com/npgsql/npgsql/blob/a43058a4ace95e4a74fe22a7fa4d77e6787d6cf3/src/Npgsql/TypeHandlers/DateTimeHandlers/TimestampTzHandler.cs#L55

And see here, where it interprets DateTimeKind.Unspecified as Utc:
https://github.com/npgsql/npgsql/blob/a43058a4ace95e4a74fe22a7fa4d77e6787d6cf3/src/Npgsql/TypeHandlers/DateTimeHandlers/TimestampTzHandler.cs#L96
